mlrun/launcher/base.py

```diff
@@ -187,13 +189,45 @@ def _validate_run_params(self, parameters: dict[str, Any]):
             )
 
     @classmethod
-    def _validate_run_single_param(cls, param_name, param_value):
+    def _validate_run_single_param(cls, param_name: str, param_value: int):
         # verify that integer parameters don't exceed a int64
         if isinstance(param_value, int) and abs(param_value) >= 2**63:
             raise mlrun.errors.MLRunInvalidArgumentError(
                 f"Parameter {param_name} value {param_value} exceeds int64"
             )
 
+    @staticmethod
+    def _validate_retry(runtime_kind: str, retry: Optional["mlrun.model.Retry"]):
+        if retry is None or not retry.count:
```

Comment:
why is the retry on base and not on server side launcher? because local=True kind of jobs?

Reply:
good point, perhaps better to move to server side and it will be ignored for local=True runs